tests/infra/logging_app.py

```diff
@@ -69,37 +83,39 @@ def issue(
 
         network.wait_for_node_commit_sync()
 
-    def verify(self, network, timeout=3):
+    def verify(self, network=None, node=None, timeout=3):
         LOG.info("Verifying all logging txs")
-
-        remote_node, _ = network.find_primary()
-        for pub_idx, pub_value in self.pub.items():
-            # As public records do not yet handle historical queries,
-            # only verify the latest entry
-            entry = pub_value[-1]
-            self._verify_tx(
-                remote_node,
-                pub_idx,
-                entry["msg"],
-                entry["seqno"],
-                entry["view"],
-                priv=False,
-                timeout=timeout,
-            )
-
-        for priv_idx, priv_value in self.priv.items():
-            for v in priv_value:
+        if network is not None:
+            self.network = network
+        nodes = self.network.get_joined_nodes() if node is None else [node]
+        for node in nodes:
+            for pub_idx, pub_value in self.pub.items():
+                # As public records do not yet handle historical queries,
```

Comment:
Is that right? I thought private ones didn't?

Reply:
At the moment, only private records support historical queries: https://github.com/microsoft/CCF/blob/a59b09e5d9589ec755ca2935a581c7142802febf/src/apps/logging/logging.cpp#L340

Reply:
To be clear, this is just that we only have an endpoint for querying private historical entries, and none for querying public. There's nothing technical preventing us from querying public state, its just a case of code duplication/writing a more general endpoint, so we haven't done it yet.